applications/ExaquteSandboxApplication/custom_processes/divergence_process.cpp

```diff
@@ -0,0 +1,180 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Tosi
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+
+// Application includes
+#include "custom_processes/divergence_process.h"
+
+
+namespace Kratos
+{
+    /* Public functions *******************************************************/
+
+    // Constructor
+    DivergenceProcess::DivergenceProcess(
+        ModelPart& rModelPart):
+        Process(),
+        mrModelPart(rModelPart)
+    {}
+
+    std::string DivergenceProcess::Info() const
+    {
+        return "DivergenceProcess";
+    }
+
+    void DivergenceProcess::PrintInfo(std::ostream& rOStream) const
+    {
+        rOStream << "DivergenceProcess";
+    }
+
+    void DivergenceProcess::PrintData(std::ostream& rOStream) const
+    {
+        this->PrintInfo(rOStream);
+    }
+
+    void DivergenceProcess::ExecuteInitialize()
+    {
+        KRATOS_TRY;
+
+        auto& r_elements_array = mrModelPart.Elements();
+        // Initialize variables
+        VariableUtils().SetNonHistoricalVariableToZero(DIVERGENCE, r_elements_array);
+        VariableUtils().SetNonHistoricalVariableToZero(VELOCITY_H1_SEMINORM, r_elements_array);
+
+        KRATOS_CATCH("");
+    }
+
+    void DivergenceProcess::ExecuteBeforeOutputStep()
+    {
+        KRATOS_TRY;
+
+        // Extract information
+        const std::size_t dimension = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+        // Check and set number of elements and check dimension
+        KRATOS_ERROR_IF(dimension < 2 || dimension > 3) << "Inconsinstent dimension to execute the process. Dimension =" << dimension << " but needed dimension = 2 or dimension = 3." << std::endl;
+        KRATOS_ERROR_IF(mrModelPart.NumberOfElements() == 0) << "The number of elements in the domain is zero. The process can not be applied."<< std::endl;
+        const unsigned int number_elements = mrModelPart.NumberOfElements();
+
+        // Auxiliar containers
+        GeometryData::ShapeFunctionsGradientsType DN_DX;
+
+        // Iterate over the elements
+        #pragma omp parallel for firstprivate(DN_DX)
+        for(int i_elem = 0; i_elem < static_cast<int>(number_elements); ++i_elem) {
+            auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+            auto& r_geometry = it_elem->GetGeometry();
```

Comment:
At first glance, I think this can be `const`.
```suggestion
            const auto& r_geometry = it_elem->GetGeometry();
```

Reply:
Done.